homeassistant/components/switch/tuya.py

```diff
@@ -0,0 +1,73 @@
+"""
+Simple platform to control **SOME** Tuya switch devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.tuya/
+"""
+import voluptuous as vol
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_HOST, CONF_ID)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pytuya==1.0']
+
+CONF_DEVICE_ID = 'device_id'
+CONF_LOCAL_KEY = 'local_key'
+
+DEFAULT_ID = 1
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_DEVICE_ID): cv.string,
+    vol.Required(CONF_LOCAL_KEY): cv.string,
+    vol.Optional(CONF_ID, default=DEFAULT_ID): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of the Tuya switch."""
+    import pytuya
+
+    add_devices([TuyaDevice(
+        pytuya.OutletDevice(
+            config.get(CONF_DEVICE_ID),
+            config.get(CONF_HOST),
+            config.get(CONF_LOCAL_KEY),
+        ),
+        config.get(CONF_NAME),
+        config.get(CONF_ID)
+    )])
+
+
+class TuyaDevice(SwitchDevice):
+    """Representation of a Tuya switch."""
+
+    def __init__(self, device, name, switchid):
+        """Initialize the Tuya switch."""
+        self._device = device
+        self._name = name
+        self._state = False
+        self._switchid = switchid
+
+    @property
+    def name(self):
+        """Get name of Tuya switch."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Check if Tuya switch is on."""
+        return self._state
+
+    def turn_on(self, **kwargs):
+        """Turn Tuya switch on."""
+        self._device.set_status(True, self._switchid)
+        self._state = True
+        self.async_schedule_update_ha_state()
```

Comment:
Remove this. It will be called by the service handler after this method (`turn_on`) is called.

Reply:
The underlying library supports polling via status() but for whatever reason this fails a lot when running it from HomeAssistant. It seems to work 100% from my desktop but when running in HA the tuya devices forcibly disconnects after HA sends a request for status *most of the time*. Every so often a request will go through but usually they fail. Trying to debug at the moment

Reply:
Are there multiple switches on the same device? I think HA is polling the same device multiple times in a row and the device can't respond quick enough. (My device has 6 switches so HA polls the same device 6 times in a row.)